File: security/demo-api-with-spring-security/src/main/java/io/costax/demo/api/BooksController.java
package io.costax.demo.api;


import io.costax.demo.domain.model.Book;
import io.costax.demo.domain.repositories.BookRepository;
import io.costax.demo.domain.repositories.BookSpecifications;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping(
        value = "/books",
        produces = {
                MediaType.APPLICATION_JSON_VALUE
        })
public class BooksController {

    private final BookRepository bookRepository;

    public BooksController(final BookRepository bookRepository) {
        this.bookRepository = bookRepository;
    }

    @GetMapping
    public List<Book> list() {
        return bookRepository.findAll();
    }

    @ResponseStatus(HttpStatus.OK)
    @GetMapping("/{id}")
    public ResponseEntity<?> findById(@PathVariable("id") Integer id) {
        return bookRepository.findById(id)
                .map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.notFound().build());
    }

    @ResponseStatus(HttpStatus.OK)
    @GetMapping("/i/{isbn}")
    public ResponseEntity<?> findByIsbn(@PathVariable("isbn") String isbn) {
        return bookRepository.findOne(BookSpecifications.withISBN(isbn))
                .map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.notFound().build());
    }

}

File: spring-boot-tipps-graphql/firststep/src/main/kotlin/io/github/jlmc/firststep/api/controllers/BooksController.kt
package io.github.jlmc.firststep.api.controllers

import io.github.jlmc.firststep.api.models.BookResource
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping(path = ["/books"])
class BooksController {
    @GetMapping
    fun books(): List<BookResource> {
        return listOf(BookResource(1L, "book 1"), BookResource(2L, "book 2"), BookResource(3L, "book 3"))
    }
}
